internal/kibana/dashboards.go

```diff
@@ -19,8 +23,77 @@ type exportedType struct {
 	Objects []common.MapStr `json:"objects"`
 }
 
-// Export method exports selected dashboards using the Kibana Export API.
+// Export method exports selected dashboards using the Kibana APIs.
 func (c *Client) Export(dashboardIDs []string) ([]common.MapStr, error) {
+	versionInfo, err := c.Version()
+	if err != nil {
+		return nil, fmt.Errorf("could not get Kibana version info: %w", err)
+	}
+
+	sv, err := semver.NewVersion(versionInfo.Number)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse Kibana version: %w", err)
+	}
+
+	if sv.LessThan(semver.MustParse("8.8.0")) {
+		return c.exportWithDashboardsAPI(dashboardIDs)
+	}
+
+	return c.exportWithSavedObjectsAPI(dashboardIDs)
+}
+
+type exportSavedObjectsRequest struct {
+	ExcludeExportDetails  bool                              `json:"excludeExportDetails"`
+	IncludeReferencesDeep bool                              `json:"includeReferencesDeep"`
+	Objects               []exportSavedObjectsRequestObject `json:"objects"`
+}
+
+type exportSavedObjectsRequestObject struct {
+	ID   string `json:"id"`
+	Type string `json:"type"`
+}
+
+func (c *Client) exportWithSavedObjectsAPI(dashboardIDs []string) ([]common.MapStr, error) {
+	logger.Debug("Export dashboards using the Kibana Saved Objects Export API")
+
+	exportRequest := exportSavedObjectsRequest{
+		ExcludeExportDetails: true,
```

Comment:
Would it be needed to set `IncludeReferencesDeep` to true, to get other references used in the dashboards like tags, searches, maps... ?

Reply:
Not sure, trying to add it.